pkg/controller/pxc/users.go

```diff
@@ -0,0 +1,284 @@
+		case "monitor":
+			*restartProxy = true
+			hosts = []string{"10.%", "%"}
+		case "clustercheck":
+			*restartProxy = true
+			*restartPXC = true
+			hosts = []string{"localhost"}
+		case "proxyadmin":
+			*restartProxy = true
+			continue
+
+		case "pmmserver":
+			if cr.Spec.PMM.Enabled {
+				*restartProxy = true
+				*restartPXC = true
+				continue
+			}
+		}
+		user := users.SysUser{
+			Name:  name,
+			Pass:  string(pass),
+			Hosts: hosts,
+		}
+		sysUsers = append(sysUsers, user)
+	}
+
+	if len(sysUsers) > 0 {
+		err = um.UpdateUsersPass(sysUsers)
+		if err != nil {
+			return errors.Wrap(err, "update sys users pass")
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) restartPXC(cr *api.PerconaXtraDBCluster, newSecretDataHash string, sysUsersSecretObj *corev1.Secret) error {
+	sfsPXC := appsv1.StatefulSet{}
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      cr.Name + "-pxc",
+		},
+		&sfsPXC,
+	)
+
+	if len(sfsPXC.Annotations) == 0 {
+		sfsPXC.Annotations = make(map[string]string)
+	}
+	sfsPXC.Spec.Template.Annotations["last-applied-secret"] = newSecretDataHash
+
+	err = r.client.Update(context.TODO(), &sfsPXC)
+	if err != nil {
+		return errors.Wrap(err, "update pxc sfs last-applied annotation")
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) restartProxy(cr *api.PerconaXtraDBCluster, newSecretDataHash string) error {
+	pvcProxy := corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      "proxydata-" + cr.Name + "-proxysql-0",
+		},
+	}
+	err := r.client.Delete(context.TODO(), &pvcProxy)
+	if err != nil {
+		return errors.Wrap(err, "delete proxy pvc")
+	}
+
+	sfsProxy := appsv1.StatefulSet{}
+	err = r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      cr.Name + "-proxysql",
+		},
+		&sfsProxy,
+	)
+	if err != nil {
+		return errors.Wrap(err, "get proxy statefulset")
+	}
+
+	err = r.client.Delete(context.TODO(), &sfsProxy)
+	if err != nil {
+		return errors.Wrap(err, "delete proxy statefulset")
+	}
+
+	return nil
+}
+
+// getInternalSysUsersSecret return secret created by operator for storing system users data
+func (r *ReconcilePerconaXtraDBCluster) getInternalSysUsersSecret(cr *api.PerconaXtraDBCluster, sysUsersSecretObj *corev1.Secret) (corev1.Secret, error) {
+	secretName := internalPrefix + cr.Spec.SecretsName
+	internalSysUsersSecretObj, err := r.getInternalSysUsersSecretObj(cr, sysUsersSecretObj)
+	if err != nil {
+		return internalSysUsersSecretObj, errors.Wrap(err, "create internal sys users secret object")
+	}
+	err = r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      secretName,
+		},
+		&internalSysUsersSecretObj,
+	)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return internalSysUsersSecretObj, errors.Wrap(err, "get internal sys users secret")
+	} else if k8serrors.IsNotFound(err) {
```

Comment:
else is not needed

Reply:
fixed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e